python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -53,6 +53,7 @@
 
 class DlgSqlWindow(QWidget, Ui_Dialog):
     nameChanged = pyqtSignal(str)
+    QUERY_HISTORY_LIMIT = 20
```

Comment:
does it make sense to have the rolling buffer configuration size in the QgsSettings (advanced).
I can't say for a common user, so I would leave 20 as fixed value.
You have much more experience than me with this, so try to think if there are class of use cases where the user need a longer or shorter history.

Reply:
As it is now, users have 20 entries for each database. I think that it is good, so we can try to merge and waiting for any feedback from users and then reworking on it.